### PR TITLE
Add upgrade kubernetes procedure for oc

### PR DESCRIPTION
**Pull request by soltysh**
/assign @mfojtik @damemi @sallyom @tnozicka @ingvagabund 
/cc @deads2k

### CODE REVIEW

File: enhancements/oc/upgrade_kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Upgrading oc to latest version of Kubernetes with gomod
+
+1. Explicitly opt into go modules, even though we are inside a `GOPATH` directory,
+   otherwise the default auto mechanism turns on, which when it notices vendor
```
Comment (abhinavdahiya) [inline enhancements/oc/upgrade_kubernetes.md]: > `default auto mechanism turns on, which when it notices vendor`

Do you have some upstream doc that provides details on this behavior, i have been looking for something like that but haven't found anything :(

File: enhancements/oc/upgrade_kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   directory will turn go modules off.
+   ```
+   export GO111MODULE=on
+   ```
+2. Get desired version of apimachinery, client-go, cli-runtime, kubectl and kubernetes (eg. 1.16.0).
+3. Create branch `oc-A.B-kubernetes-X.Y.Z` (where A.B is the oc version, and X.Y.Z is the kubernetes version)
+   in apimachinery, client-go, cli-runtime, kubectl and kubernetes and prime the repos with the basic
+   state from k8s (see previous step).
+4. Pick carry patches for appropriate repositories. `git log --no-merges --oneline v1.14.0..openshift/oc-4.2-kubernetes-1.14.0`
```
Comment (abhinavdahiya) [inline enhancements/oc/upgrade_kubernetes.md]: > git log --no-merges --oneline v1.14.0..openshift/oc-4.2-kubernetes-1.14.0

Nice

File: enhancements/oc/upgrade_kubernetes.md (hunk 1 above)
Comment (soltysh) [reply enhancements/oc/upgrade_kubernetes.md]: https://github.com/golang/go/wiki/Modules#recent-changes is the closest, but that is only for the time of doing the bump. We'll eventually want to turn them off explicitly to ensure we use `vendor` dir contents.

**Pull request merged**